### PR TITLE
Create loading state for save button on ProjectForm.js

### LINKED ISSUE #1574: Create loading state for save buttons on forms related to projects
### Dependency
This issue should be delayed until https://github.com/hackforla/VRMS/issues/1576 is solved due to potential merge conflicts

### Overview
Set save button to spinning loading icon while waiting for database to reply

### Action Items
- [x] Edit `ProjectForm` component save button to be disabled and show a loading icon to the user.
  - [x] Model this state off of the `ReadyEvents` file
  - [x] Update `submitNewProject` and `submitEditProject` to initialized loading state when called and end loading state when returned or caught error.
 - [x] QA
 - [x] When complete, go to #1598 and mark the checkbox referencing this issue as complete.

### PR DESCRIPTION
**Pull request by lcchrty**
Fixes #1574 

### What changes did you make and why did you make them ?

  - Added loading state
  - Conditionally rendered disabled save button based on load state
  - updated dependency arrays in `ManageProjects.js` to eliminate compile errors

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image

</details>

### CODE REVIEW

**Review by trillium — COMMENTED** [review]
Gave some notes, leme know what you think!

File: client/src/components/ProjectForm.js [hunk 1]
```diff
@@ -266,17 +274,39 @@ export default function ProjectForm({
         <Box>
           <Grid container justifyContent="space-evenly" sx={{ my: 3 }}>
             <Grid item xs="auto">
-              <StyledButton
```
Comment (trillium) [inline client/src/components/ProjectForm.js]: Hey hey

I think this section based on isLoading looks like a duplication of everything except the child element.

How do you feel about something like this instead?

```js
              <StyledButton
                type="submit"
                form="project-form"
                variant={
                  !isEdit ? 'secondary' : !editMode ? 'contained' : 'secondary'
                }
                cursor="pointer"
                disabled={isEdit ? !editMode : false}
              >
                { isLoading ? <CircularProgress /> : Save }
              </StyledButton>

**Review by trillium — APPROVED** [review]
Yes! Thanks, this is looking good :)

**Comment by JackHaeg** [comment]
@Spiteless Just checking in! I noticed that this PR is still open, and it looks like it's ready to be merged. If it's ready to go, can we merge & close the issue?

**Pull request merged**